File: Python/Dict_example.py
def main() -> None:
    user = {
        "name": "Daniel",
        "age": 25,
        "country": "USA",
    }

    print("Original:", user)

    # get
    print('get("name"):', user.get("name"))
    print('get("email", "not found"):', user.get("email", "not found"))

    # keys, values, items
    print("keys():", list(user.keys()))
    print("values():", list(user.values()))
    print("items():", list(user.items()))

    # update
    user.update({"age": 26, "email": "daniel@example.com"})
    print("after update():", user)

    # setdefault
    user.setdefault("city", "New York")
    user.setdefault("name", "Other Name")  # won't overwrite existing
    print("after setdefault():", user)

    # pop
    removed_email = user.pop("email")
    print("popped email:", removed_email)
    print('after pop("email"):', user)

    # popitem (removes last inserted pair in Python 3.7+)
    last_item = user.popitem()
    print("popitem():", last_item)
    print("after popitem():", user)

    # copy
    user_copy = user.copy()
    print("copy():", user_copy)

    # clear (on copy only, so original stays)
    user_copy.clear()
    print("after clear() on copy:", user_copy)
    print("original still:", user)

    # fromkeys
    default_scores = dict.fromkeys(["math", "science", "english"], 0)
    print("fromkeys():", default_scores)


if __name__ == "__main__":
    main()

File: Javascript/Dict_example.js
const main = () => {
  const user = {
    name: 'Daniel',
    age: 25,
    country: 'USA',
  };

  console.log('Original:', user);

  console.log('get("name"):', user.name);
  console.log('get("email", "not found"):', user.email ?? 'not found');

  console.log('keys():', Object.keys(user));
  console.log('values():', Object.values(user));
  console.log('items():', Object.entries(user));

  Object.assign(user, { age: 26, email: 'daniel@example.com' });
  console.log('after update():', user);

  if (!Object.hasOwn(user, 'city')) {
    user.city = 'New York';
  }
  if (!Object.hasOwn(user, 'name')) {
    user.name = 'Other Name';
  }
  console.log('after setdefault():', user);

  const removedEmail = user.email;
  delete user.email;
  console.log('popped email:', removedEmail);
  console.log('after pop("email"):', user);

  const keys = Object.keys(user);
  const lastKey = keys[keys.length - 1];
  const lastValue = user[lastKey];
  delete user[lastKey];
  console.log('popitem():', [lastKey, lastValue]);
  console.log('after popitem():', user);

  const userCopy = { ...user };
  console.log('copy():', userCopy);

  for (const key of Object.keys(userCopy)) {
    delete userCopy[key];
  }
  console.log('after clear() on copy:', userCopy);
  console.log('original still:', user);

  const defaultScores = Object.fromEntries(
    ['math', 'science', 'english'].map((subject) => [subject, 0]),
  );
  console.log('fromkeys():', defaultScores);
};

main();
